airbyte-webapp/src/hooks/useRouter.tsx

```diff
@@ -1,39 +1,19 @@
 import queryString from "query-string";
 import { useMemo } from "react";
-import { useLocation, useNavigate, useParams, Location, To, NavigateOptions } from "react-router-dom";
+import { useLocation } from "react-router-dom";
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function useRouter<T = any, P = any>(): {
-  query: T;
-  params: P;
-  pathname: string;
-  location: Location;
-  push(path: To, state?: NavigateOptions): void;
-  replace(path: To, state?: NavigateOptions): void;
-} {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const params: any = useParams();
-  const location = useLocation();
-  const navigate = useNavigate();
-  const query = useMemo<T>(
-    () =>
-      ({
-        ...queryString.parse(location.search), // Convert string to object
-        ...params,
-      } as T),
-    [params, location.search]
-  );
+import { SortOrderEnum } from "../components/EntityTable/types";
 
-  return useMemo(() => {
-    return {
-      params,
-      push: navigate,
-      replace: (path, state) => navigate(path, { ...state, replace: true }),
-      pathname: location.pathname,
-      query,
-      location,
-    };
-  }, [navigate, location, query, params]);
+interface UseQueryResult {
+  from?: string;
+  id?: string;
+  mode?: string;
+  oobCode?: string;
+  order?: SortOrderEnum;
+  sortBy?: string;
```

Comment:
Putting this into this central place does not feel like the best TypeScript solution. I'd suggest we instead make `useQuery` to take a generic parameter, which actually then will be the return type, so that every caller of `useQuery` will specify the parameters it knows exist on this page.

If we want to make it more "fail safe" here, and say we want to make sure that each of the parameters could still be undefined, we could still put this into the return type, i.e.

```ts
useQuery = <T>(): Partial<T> => {
```